pkg/operator/controller/route-status/route_status_test.go

```diff
@@ -0,0 +1,469 @@
+			routes: routev1.RouteList{
+				Items: []routev1.Route{
+					*newRouteWithLabelWithAdmittedStatuses("foo-route", "ns", "shard-label", "foo-ic", "bar-ic"),
+					*newRouteWithLabelWithAdmittedStatuses("bar-route", "ns2", "", "baz-ic", "foo-ic", "bar-ic"),
+				},
+			},
+			namespace: corev1.NamespaceList{
+				Items: []corev1.Namespace{
+					*newNamespace("ns", "shard-label"),
+					*newNamespace("ns2", ""),
+				},
+			},
+			ingressController: newIngressControllerWithSelectors("foo-ic", "shard-label", "", "shard-label", ""),
+			expectedRoutes: routev1.RouteList{
+				Items: []routev1.Route{
+					*newRouteWithLabelWithAdmittedStatuses("foo-route", "ns", "shard-label", "foo-ic", "bar-ic"),
+					*newRouteWithLabelWithAdmittedStatuses("bar-route", "ns", "", "baz-ic", "bar-ic"),
+				},
+			},
+		},
+		{
+			name: "clear route that is no longer admitted by ingress controller by namespaceSelectors expression",
+			routes: routev1.RouteList{
+				Items: []routev1.Route{
+					*newRouteWithLabelWithAdmittedStatuses("foo-route", "ns", "shard-label", "foo-ic", "bar-ic"),
+					*newRouteWithLabelWithAdmittedStatuses("bar-route", "ns2", "", "baz-ic", "foo-ic", "bar-ic"),
+				},
+			},
+			namespace: corev1.NamespaceList{
+				Items: []corev1.Namespace{
+					*newNamespace("ns", "shard-label"),
+					*newNamespace("ns2", ""),
+				},
+			},
+			ingressController: newIngressControllerWithSelectors("foo-ic", "shard-label", "", "", "shard-label"),
+			expectedRoutes: routev1.RouteList{
+				Items: []routev1.Route{
+					*newRouteWithLabelWithAdmittedStatuses("foo-route", "ns", "shard-label", "foo-ic", "bar-ic"),
+					*newRouteWithLabelWithAdmittedStatuses("bar-route", "ns", "", "baz-ic", "bar-ic"),
+				},
+			},
+		},
+		{
+			name:              "nil ingress controller",
+			routes:            routev1.RouteList{},
+			namespace:         corev1.NamespaceList{},
+			ingressController: nil,
+			expectedRoutes:    routev1.RouteList{},
+			expectedErr:       true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Load up client with test objects.
+			client := unit.NewFakeClient()
+			if tc.ingressController != nil {
+				if err := client.Create(context.Background(), tc.ingressController); err != nil {
+					t.Errorf("error ingress controller route: %v", err)
+				}
+			}
+			for _, route := range tc.routes.Items {
+				if err := client.Create(context.Background(), &route); err != nil {
+					t.Errorf("error creating route: %v", err)
+				}
+			}
+			for _, ns := range tc.namespace.Items {
+				if err := client.Create(context.Background(), &ns); err != nil {
+					t.Errorf("error creating ns: %v", err)
+				}
+			}
+
+			if errs := ClearRoutesNotAdmittedByIngress(client, tc.ingressController); tc.expectedErr && len(errs) == 0 {
+				t.Errorf("expected errors, got no errors")
+			} else if !tc.expectedErr && len(errs) != 0 {
+				t.Errorf("did not expected errors: %v", errs)
```

Comment:
These can probably all use `t.Fatal` or `t.Fatalf` instead of `t.Errorf`.

Reply:
Done